CHICKEN docs and updates (#134514)
NixOS/nixpkgs: doc/languages-frameworks/chicken.section.md

> # CHICKEN {#sec-chicken}

Author: I still feel like that can be done later, when a Guile or Racket or etc. section is written. In particular, I don't think that it's a great idea to group languages based on linguistic or PLT similarity instead of toolchain similarity; at least when I'm in this section of the docs, it's usually in order to figure out how to do something relatively specific but well-documented in generic terms, and I want to instruct a specific toolchain in a specific way.

> # CHICKEN {#sec-chicken}

Reviewer: Thought about this again, and given that the TOC hides sub sections, it may make chicken less discoverable, so let's keep it as is for now?!

[CHICKEN](https://call-cc.org/) is a
[R⁵RS](https://schemers.org/Documents/Standards/R5RS/HTML/)-compliant Scheme
compiler. It includes an interactive mode and a custom package format, "eggs".

## Using Eggs

Eggs described in nixpkgs are available inside the
`chickenPackages.chickenEggs` attrset. Including an egg as a build input is
done in the typical Nix fashion. For example, to include support for [SRFI
189](https://srfi.schemers.org/srfi-189/srfi-189.html) in a derivation, one

> might write:

Reviewer: Is there anything special how this works? E. g. are there any chicken-specific environment variables and setup hooks involved?

```nix
  buildInputs = [
    chicken
    chickenPackages.chickenEggs.srfi-189
  ];
```

## Updating Eggs

> nixpkgs does not know about all eggs, but includes support for
> [egg2nix](https://github.com/the-kenny/egg2nix), which can be used to
> automatically include published eggs.

Reviewer: I think it's a bit clearer like this (or similar):

```suggestion
nixpkgs only knows about a subset of all published eggs. It uses
[egg2nix](https://github.com/the-kenny/egg2nix) to generate a
package set from a list of eggs to include.
```

> The procedure consists of running the following shell commands:

Reviewer: Does this only regenerate the package set? Or does it also update the versions involved? Are the versions pinned?

```suggestion
The package set is regenerated by running the following shell commands:
```

Author: Versions are pinned. We take an all-or-nothing snapshot of working packages, akin to our Node.js integration. This is probably what we desire, but it does mean that introducing new eggs must happen after we update the existing eggs. I'll add this to the doc, too.
